scipy/stats/tests/test_morestats.py

```diff
@@ -2975,90 +2980,96 @@ def test_directional_stats_correctness(self):
-        assert_allclose(res, ref)
+        xp_assert_close(res, ref)
 
-    def test_directional_stats_2d(self):
+    def test_directional_stats_2d(self, xp):
         # Test that for circular data directional_stats
         # yields the same result as circmean/circvar
         rng = np.random.default_rng(0xec9a6899d5a2830e0d1af479dbe1fd0c)
-        testdata = 2 * np.pi * rng.random((1000, ))
-        testdata_vector = np.stack((np.cos(testdata),
-                                    np.sin(testdata)),
+        testdata = xp.asarray(2 * xp.pi * rng.random((1000, )))
+        testdata_vector = xp.stack((xp.cos(testdata),
+                                    xp.sin(testdata)),
                                    axis=1)
         dirstats = stats.directional_stats(testdata_vector)
         directional_mean = dirstats.mean_direction
-        directional_mean_angle = np.arctan2(directional_mean[1],
-                                            directional_mean[0])
-        directional_mean_angle = directional_mean_angle % (2*np.pi)
+        xp_test = array_namespace(directional_mean)  # np needs atan2
+        directional_mean_angle = xp_test.atan2(directional_mean[1],
+                                               directional_mean[0])
+        directional_mean_angle = directional_mean_angle % (2 * xp.pi)
         circmean = stats.circmean(testdata)
-        assert_allclose(circmean, directional_mean_angle)
+        xp_assert_close(directional_mean_angle, circmean)
 
-        directional_var = 1 - dirstats.mean_resultant_length
+        directional_var = 1. - dirstats.mean_resultant_length
         circular_var = stats.circvar(testdata)
-        assert_allclose(directional_var, circular_var)
+        xp_assert_close(directional_var, circular_var)
 
-    def test_directional_mean_higher_dim(self):
+    def test_directional_mean_higher_dim(self, xp):
         # test that directional_stats works for higher dimensions
         # here a 4D array is reduced over axis = 2
         data = np.array([[0.8660254, 0.5, 0.],
                          [0.8660254, -0.5, 0.]])
-        full_array = np.tile(data, (2, 2, 2, 1))
-        expected = np.array([[[1., 0., 0.],
-                              [1., 0., 0.]],
-                             [[1., 0., 0.],
-                              [1., 0., 0.]]])
+        full_array = xp.asarray(np.tile(data, (2, 2, 2, 1)))
+        expected = xp.asarray([[[1., 0., 0.],
+                                [1., 0., 0.]],
+                               [[1., 0., 0.],
+                                [1., 0., 0.]]],
+                              dtype=xp.float64)
         dirstats = stats.directional_stats(full_array, axis=2)
-        assert_allclose(expected, dirstats.mean_direction)
+        xp_assert_close(dirstats.mean_direction, expected)
 
-    def test_directional_stats_list_ndarray_input(self):
+    def test_directional_stats_list_ndarray_input(self, xp):
```

Comment:
Looks like this can be `np_only=True` because the intent is to check behavior for lists.